backend/applications/tests/test_applications_api.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+from rest_framework.reverse import reverse
+
+from applications.api.v1.serializers import ApplicationSerializer
+from applications.enums import ApplicationStatus
+from applications.tests.factories import ApplicationFactory
+
+
+@pytest.mark.django_db
+def test_applications_list(api_client):
+    ApplicationFactory()
+    response = api_client.get(reverse("v1/applications:application-list"))
+
+    assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_application_single_read(api_client):
+    application = ApplicationFactory()
+    response = api_client.get(
+        reverse("v1/applications:application-detail", kwargs={"pk": application.id})
+    )
+
+    assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_application_put(api_client):
+    application = ApplicationFactory()
+    serializer = ApplicationSerializer(application)
+    data = serializer.data
+    data["invoicer_name"] = "test"
+    response = api_client.put(
+        reverse("v1/applications:application-detail", kwargs={"pk": application.id}),
+        data,
+        format="json",
+    )
+
+    assert response.status_code == 200
+    assert response.data["invoicer_name"] == "test"
+
+
+@pytest.mark.django_db
+def test_application_patch(api_client):
+    application = ApplicationFactory()
+    application.status = ApplicationStatus.DRAFT
+    application.save()
```

Comment:
Can't the status be passed to the factory? Like `ApplicationFactory(status=ApplicationStatus.DRAFT)` ? I might be remembering wrong here however, so please correct me in that case 😄

Reply:
Yes it can :D I'll fix.